core/src/main/scala/zio/config/ReadError.scala

```diff
@@ -37,9 +20,13 @@ sealed trait ReadError[A] extends Exception { self =>
 }
 
 object ReadError {
-  final case class MissingValue[A](path: Vector[Either[Int, A]])                     extends ReadError[A]
-  final case class FormatError[A](path: Vector[Either[Int, A]], message: String)     extends ReadError[A]
-  final case class ConversionError[A](path: Vector[Either[Int, A]], message: String) extends ReadError[A]
-  final case class OrErrors[A](list: List[ReadError[A]])                             extends ReadError[A]
-  final case class AndErrors[A](list: List[ReadError[A]])                            extends ReadError[A]
+  sealed trait Step[+K]
+  case class KeyStep[+K](key: K)   extends Step[K]
+  case class IndexStep(index: Int) extends Step[Nothing]
```

Comment:
👍 

Add `final` to both case classes here.

Reply:
Ah and put `KeyStep` and `IndexStep` into the companion object of `Step`, which will let you shorten `KeyStep` into `Key`, and `IndexStep` into `Index`.

Reply:
done